snakemake_executor_plugin_kubernetes/__init__.py

```diff
@@ -223,55 +244,64 @@ def run_job(self, job: JobExecutorInterface):
             )
             container.env.append(envvar)
 
-        # request resources
-        self.logger.debug(f"job resources:  {dict(job.resources)}")
+        # Request resources
+        self.logger.debug(f"Job resources: {dict(job.resources)}")
         container.resources = kubernetes.client.V1ResourceRequirements()
         container.resources.requests = {}
+        container.resources.limits = {}
+
+        # CPU and memory requests
         container.resources.requests["cpu"] = "{}m".format(
             int(job.resources["_cores"] * self.k8s_cpu_scalar * 1000)
         )
+        container.resources.limits["cpu"] = "{}m".format(
+            int(job.resources["_cores"] * 1000)
+        )
         if "mem_mb" in job.resources.keys():
             container.resources.requests["memory"] = "{}M".format(
                 job.resources["mem_mb"]
             )
+            container.resources.limits["memory"] = "{}M".format(
+                job.resources["mem_mb"]
+            )
         if "disk_mb" in job.resources.keys():
             disk_mb = int(job.resources.get("disk_mb", 1024))
             container.resources.requests["ephemeral-storage"] = f"{disk_mb}M"
+            container.resources.limits["ephemeral-storage"] = f"{disk_mb}M"
 
+        # Request GPU resources if specified
+        if "gpu" in job.resources.keys():
+            gpu_count = str(job.resources["gpu"])
+            container.resources.requests["nvidia.com/gpu"] = gpu_count
+            container.resources.limits["nvidia.com/gpu"] = gpu_count
+            self.logger.debug(f"Requested GPU resources: {gpu_count}")
```

Comment:
What about non-nvidia GPUs? Maybe infer this from a `gpu_model: str` resource, which could allow types like `nvidia` or more specifically `tesla` etc.?

Reply:
Well I specify the machine type already using the `machine_type="g2-standard-8`. But also I believe with GKE the node pool inherently specifies `"nvidia.com/gpu"`, without it the node scaling doesn't get triggered.

Reply:
ok, but this plugin is meant to be generic, so we should try to ensure that added functionality works with any k8s cluster